### ISSUE TITLE
DataValidation Bug

### ISSUE THREAD

**Issue by alandoyle95** [dtjohnson/xlsx-populate#90]
There is currently a bug where if a cell already has dataValidation, and you try to add it with code, it keeps the old dataValidation vs the new one...

**Comment by alandoyle95** [comment]
nevermind something was not working correctly in my code

**Comment by alandoyle95** [comment]
Confirmed.. there is a slight bug with the combination of defined names and dataValidation.

defined names with merged cells are attached to the first cell. and then the data validation from excel is attached to the range of cells that are merged. so will need to add a check to see if that cell is merged at all. and then if it is, get the range and then do the data validation.

i will work on a patch to fix this inside of xlsx-populate's code as well.

**Comment by dtjohnson** [comment]
Yeah, there is some funniness with merge cells and whether operations. A simple case is a merge cell of A1:C3. What should happen when you change the value, style, etc, of A1? What about B2? My thinking is to just leave it be and let authors deal with the structure of their own workbooks. It might be nice to have some helper methods for seeing if a cell is part of a merge range though.

**Comment by alandoyle95** [comment]
Yea I was thinking the gettter for `.merged` instead of `true or false`
would be `range` or` false`

**Comment by dtjohnson** [comment]
That's a method on Range. We'd need something new method on Cell to see if
it's part of a range.

**Comment by alandoyle95** [comment]
What if we made a function in cell.js called isMerged and make it return
the range if true and false if it's not a cell inside of a range.

**Comment by dtjohnson** [comment]
I'd prefer something named a bit more obvious like `Cell.mergedRange` that would return null if not part of a merge and the `Range` if it is. The method would need to look through all merged ranges on the cell's parent sheet and return the first range that contains the cell. To do this, you'll need to do something like:
* Add a method `Range.containsCell` that takes a `Cell` or address or row/column. It would to check if the cell's row/column is between the range's start/end cell rows and columns. Be careful though! The range start/end cells don't have to be in top-left/bottom-right order. Use the `Range._minRowNumber`, etc. private properties.
* Add a method `Sheet.mergedRanges` that would return an array of all of the merged ranges. You should just be able something like `_(this._mergeCells).keys().map(adddress => this.range(address)).value()`.
* Add a method `Cell.mergedRange` that would just call `_.find(this.sheet().mergedRanges(), range => range.containsCell(this))`.
